src/sentry/filestore/s3.py

```diff
@@ -300,6 +314,7 @@ def __init__(self, acl=None, bucket=None, **settings):
         self._entries = {}
         self._bucket = None
         self._connection = None
+        self._lock = threading.Lock()
```

Comment:
Surely we can kill the lock entirely now since you cannot actually concurrently use it since the session is thread bound.